File: readability C/readability.c
#include<cs50.h>
#include<stdio.h>
#include<unistd.h>
#include<string.h>
#include<math.h>

int main(void)
{
    float wordCount = 0;
    float sentenceCount = 0;
    int i, a;
    string text = get_string("Text: ");
    float numOfLetters = strlen(text);

    for (i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            wordCount++;
            numOfLetters--;

        }

    }
    for (a = 0; text[a] != '\0'; a++)
    {
        if (text[a] == '.' && text[a + 1] != '.')
        {
            sentenceCount++;
            numOfLetters--;
        }
        if (text[a] == ',' && text[a + 1] != ',')
        {
            numOfLetters--;
        }
        if (text[a] == '!' && text[a + 1] != '!')
        {
            sentenceCount++;
            numOfLetters--;
        }
        if (text[a] == '?' && text[a + 1] != '?')
        {
            sentenceCount++;
            numOfLetters--;
        }
        if (text[a] == '\'' && text[a + 1] != '\'')
        {
            numOfLetters--;
        }
        if (text[a] == '\"' && text[a + 1] != '\"')
        {
            numOfLetters--;
        }
    }
    wordCount++;
    float L = (numOfLetters / wordCount) * 100;
    float S = (sentenceCount / wordCount) * 100;
    int indexRound = round(0.0588 * L - 0.296 * S - 15.8);
    if (indexRound < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (indexRound > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", indexRound);


    }
}

File: readability py/readability.py

# anouncing all of the variables
text = str(input("Text: "))
words = text.split()
wordCount = len(words)
textLen = len(text)
letterCount = 0
sentenceCount = 0

# finding how many of them are actually letters and the number of sentences
i = 0
while i < textLen:
    if text[i].isalpha() == True:
        letterCount += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentenceCount += 1
    i += 1

L = (letterCount / wordCount) * 100
S = (sentenceCount / wordCount) * 100

CLI = round(0.0588 * L - 0.296 * S - 15.8)
if CLI > 16:
    print("Grade 16+")
elif CLI < 1:
    print("Before Grade 1")
else:
    print("Grade: ", CLI)